src/librustdoc/clean/types.rs

```diff
@@ -366,6 +367,27 @@ crate struct Item {
     crate cfg: Option<Arc<Cfg>>,
 }
 
+impl fmt::Debug for Item {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let alternate = f.alternate();
+        // hand-picked fields that don't bloat the logs too much
+        let mut fmt = f.debug_struct("Item");
+        fmt.field("name", &self.name)
+            .field("visibility", &self.visibility)
+            .field("def_id", &self.def_id);
+        // allow printing the full item if someone really wants to
+        if alternate {
+            fmt.field("attrs", &self.attrs).field("kind", &self.kind).field("cfg", &self.cfg);
+        } else {
+            fmt.field("kind", &self.type_());
+            if let Some(docs) = self.doc_value() {
+                fmt.field("docs", &docs);
+            }
```

Comment:
Maybe emit something along the lines of `docs: None` when there are no docs to make it explicit, so people don't think the docs are just not being shown in the Debug output?